csvqb/csvqb/tests/unit/cube/qb/test_cubeqb_errorvalidation.py

```diff
@@ -111,10 +120,329 @@ def test_existing_dimension_output_uri_template():
+    measure dimension.
+    """
+    data = pd.DataFrame(
+        {
+            "Some Dimension": ["a", "b", "c"],
+            "Value": [1, 2, 3],
+        }
+    )
+
+    cube = Cube(
+        CatalogMetadata("Some Qube"),
+        data,
+        [
+            QbColumn(
+                "Some Dimension",
+                NewQbDimension.from_data("Some Dimension", data["Some Dimension"]),
+            ),
+            QbColumn(
+                "Value",
+                QbMultiMeasureObservationValue(unit=NewQbUnit("Some New Unit 1")),
+            ),
+        ],
+    )
+    errors = validate_qb_component_constraints(cube)
+
+    assert_num_validation_errors(errors, 1)
+    error = errors[0]
+    assert isinstance(error, WrongNumberComponentsError)
+    assert error.component_type == QbMultiMeasureDimension
+    assert error.expected_number == 1
+    assert error.actual_number == 0
     assert (
-        "'Existing Dimension' - an ExistingQbDimension must have an output_uri_template defined."
-        in validation_errors.message
+        error.additional_explanation
+        == "A multi-measure cube must have a measure dimension."
+    )
+
+
+def test_multi_measure_obs_val_with_multiple_measure_dimensions():
+    """
+    Ensure that a user gets an error if they try to define more than one measure dimension.
+    """
+    data = pd.DataFrame(
+        {
+            "Some Dimension": ["a", "b", "c"],
+            "Measure Dimension 1": ["A1 Measure", "B1 Measure", "C1 Measure"],
+            "Measure Dimension 2": ["A2 Measure", "B2 Measure", "C2 Measure"],
+            "Value": [1, 2, 3],
+        }
+    )
+
+    cube = Cube(
+        CatalogMetadata("Some Qube"),
+        data,
+        [
+            QbColumn(
+                "Some Dimension",
+                NewQbDimension.from_data("Some Dimension", data["Some Dimension"]),
+            ),
+            QbColumn(
+                "Measure Dimension 1",
+                QbMultiMeasureDimension.new_measures_from_data(
+                    data["Measure Dimension 1"]
+                ),
+            ),
+            QbColumn(
+                "Measure Dimension 2",
+                QbMultiMeasureDimension.new_measures_from_data(
+                    data["Measure Dimension 2"]
+                ),
+            ),
+            QbColumn(
+                "Value",
+                QbMultiMeasureObservationValue(unit=NewQbUnit("Some New Unit 1")),
+            ),
+        ],
+    )
+    errors = validate_qb_component_constraints(cube)
+
+    assert_num_validation_errors(errors, 1)
+    error = errors[0]
+    assert isinstance(error, MaxNumComponentsExceededError)
+    assert error.component_type == QbMultiMeasureDimension
+    assert error.maximum_number == 1
+    assert error.actual_number == 2
+
+
+def test_measure_dimension_with_single_measure_obs_val():
+    """
+    Ensure that when a user defines a measure dimension with a single-measure observation value, they get an error.
+    """
```

Comment:
Why is this an error? It's verbose but is it wrong?

Reply:
I see where you're coming from, however, I think they've fundamentally misunderstood something if they're taking this approach so I want to push back on them doing this. 

Don't forget that all of these validation errors are strictly advisory and don't stop the user actually doing anything.